toolchain/check/testdata/where_expr/dot_self_index.carbon

```diff
@@ -18,34 +15,28 @@ interface Empty(W:! type) {
 // T has index 0
 // .Self has index invalid, not 1
 // V has index 1, does not match .Self
+//@dump-sem-ir-begin
 fn H(T:! type, U: Empty(T) where .A = T*, V:! type) {}
 
 fn G(U: Empty(i32) where .A = i32*) {
+  //@dump-sem-ir-end
```

Comment:
I think `G` exists to validate that it type checks, probably don't need to include it in the SemIR (and if we did include it, the important part of it is the call to `H`, not the signature of `G`).
```suggestion
//@dump-sem-ir-end

fn G(U: Empty(i32) where .A = i32*) {
```